sim/deathknight/runespell.go

```diff
@@ -71,18 +70,8 @@ func (rs *RuneSpell) castInternal(sim *core.Simulation, target *core.Unit) bool
 		return result
 	}
 
-	rs.dk.LastCast = rs
-
 	rs.DoCost(sim)
 
-	if rs.onCast != nil {
-		rs.onCast(sim)
-	}
-
-	if rs.dk.BladeBarrierAura.IsActive() {
-		rs.dk.BladeBarrierAura.Refresh(sim)
-	}
```

Comment:
BladeBarrier needs to be refreshed on every DK spell cast while there are 0 blood runes. I'm not sure it had the 0 runes check before but it should be checked on every cast which i guess makes sense to be kept here

Reply:
Added this back now